core/common/src/main/java/alluxio/resource/Pool.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.resource;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Interface representing a pool of resources to be temporarily used and returned.
```

Comment:
I don't think it is a good place to document this because it is very obvious that io.netty.channel.ChannelPool is only used for netty channel pool. But the Pool interface here is meant to be used by all sorts of pooling implementation (e.g. a pool of locks). I have added a comment in NettyChannelPool and hopefully that can make it clearer.